addons/hr_skills_survey/tests/test_certification_flow.py

```diff
@@ -0,0 +1,101 @@
+            'certification': True,
+            'certification_validity_months': 3,
+            'description': '<p>Description</p>',
+            'questions_layout': 'page_per_question',
+            'scoring_success_min': 85.0,
+            'scoring_type': 'scoring_without_answers',
+            'title': 'Test Certification',
+            'users_login_required': True,
+        })
+        cls.certification_q0 = cls._add_question(
+            cls, None, 'Test question ...', 'simple_choice',
+            sequence=2,
+            constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=cls.certification.id,
+            labels=[
+                {'value': 'Correct', 'is_correct': True, 'answer_score': 1.0},
+                {'value': 'Incorrect'},
+            ])
+
+    @freeze_time('2024-03-20')
+    def test_resume_line_creation(self):
+        """ Check that the resume line is correctly created upon certification completion.
+
+        As we test the method "survey.user_input._mark_done" which is called in sudo from
+        the controller, the test is executed with the admin user.
+        """
+        ResumeLine = self.env['hr.resume.line']
+        user_input_vals = {
+            'survey_id': self.certification.id,
+            'partner_id': self.user_emp.partner_id.id,
+            'user_input_line_ids': [Command.create({
+                'question_id': self.certification_q0.id,
+                'answer_type': 'suggestion',
+                'suggested_answer_id': self.certification_q0.suggested_answer_ids.ids[0]
+            })]
+        }
+        self.env['survey.user_input'].create(user_input_vals)._mark_done()
+        resume_line = ResumeLine.search([('survey_id', '=', self.certification.id)], limit=1, order='id DESC')
+        self.assertEqual(resume_line.employee_id, self.employee_emp)
+        self.assertEqual(resume_line.name, self.certification.title)
+        self.assertEqual(resume_line.description, 'Description')
+        self.assertEqual(resume_line.line_type_id, self.env.ref('hr_skills_survey.resume_type_certification'))
+        self.assertEqual(resume_line.survey_id, self.certification)
+        self.assertEqual(resume_line.date_start, fields.Date.today())
+        self.assertEqual(resume_line.date_end, fields.Date.today() + relativedelta(months=3))
+        # When redoing the same certification, the resume line is updated
+        self.certification.description = False
+        for (validity_months, expected_date_end) in (
+                (1, fields.Date.today() + relativedelta(months=1)),
+                (6, fields.Date.today() + relativedelta(months=6)),
+                (False, False),
+        ):
+            self.certification.certification_validity_months = validity_months
+            self.env['survey.user_input'].create(user_input_vals)._mark_done()
+            resume_line = ResumeLine.search([('survey_id', '=', self.certification.id)], order='id DESC')[0]
```

Comment:
Would prefer limit=1 here as well.

Actually, you could avoid setting a limit at all, and add an assertion to len == 1, this would make sure that we are updating the line instead of creating others, what do you think?